src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -126,44 +118,20 @@ static bool CallchainIsInUserSpaceInstrumentation(
       });
 }
 
-void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
-                                    const StackSamplePerfEventData& event_data) {
-  ORBIT_CHECK(listener_ != nullptr);
-  ORBIT_CHECK(current_maps_ != nullptr);
-
-  return_address_manager_->PatchSample(event_data.tid, event_data.GetRegisters()[PERF_REG_X86_SP],
-                                       event_data.GetMutableStackData(), event_data.GetStackSize());
-
-  LibunwindstackResult libunwindstack_result =
-      unwinder_->Unwind(event_data.pid, current_maps_->Get(), event_data.GetRegisters(),
-                        event_data.GetStackData(), event_data.GetStackSize());
-
-  if (libunwindstack_result.frames().empty()) {
-    // Even with unwinding errors this is not expected because we should at least get the program
-    // counter. Do nothing in case this doesn't hold for a reason we don't know.
-    ORBIT_ERROR("Unwound callstack has no frames");
-    return;
-  }
-
-  FullCallstackSample sample;
-  sample.set_pid(event_data.pid);
-  sample.set_tid(event_data.tid);
-  sample.set_timestamp_ns(event_timestamp);
-
-  Callstack* callstack = sample.mutable_callstack();
-
+void UprobesUnwindingVisitor::HandleErrorCasesAndPatchStackSample(
```

Comment:
The patching still happens in `UprobesUnwindingVisitor::Visit`. This should just be called something like `HandleStackSampleErrorCases`.
Or maybe even `SetCallstackTypeFromStackSample`, which is more precise regarding what this function is doing. In that case I would consider calling the other one `SetCallstackTypeFromCallchainAndPatch`(`Callchain`)`.